### PR TITLE
[Y2K-1589] Remove from channel nav to confirm modal

### PR DESCRIPTION
**Pull request by jzila**
cc @keybase/y2ksquad

### CODE REVIEW

File: shared/constants/teams.tsx [hunk 1]
```diff
@@ -41,6 +42,8 @@ export const addInviteWaitingKey = (teamname: Types.Teamname, value: string) =>
 export const removeMemberWaitingKey = (teamID: Types.TeamID, id: string) => `teamRemove:${teamID};${id}`
 export const addToTeamSearchKey = 'addToTeamSearch'
 export const teamProfileAddListWaitingKey = 'teamProfileAddList'
+export const removeFromChannelWaitingKey = (conversationIDKey: ChatTypes.ConversationIDKey, id: string) =>
```
Comment (buoyad) [inline shared/constants/teams.tsx:45]: prefer a more specific param name than `id`, are they always usernames?

File: shared/teams/common/selection-popup.tsx [hunk 2]
```diff
@@ -366,7 +366,17 @@ const ChannelMembersActions = ({conversationIDKey, teamID}: ChannelActionsProps)
         path: [{props: {teamID, usernames: members}, selected: 'teamAddToChannels'}],
       })
     )
-  const onRemoveFromChannel = () => console.log('onRemoveFromChannel not implemented') // TODO
+  const onRemoveFromChannel = () =>
```
Comment (buoyad) [inline shared/teams/common/selection-popup.tsx:369]: I noticed this button is showing up for #general, we should fix that.

File: shared/teams/confirm-modals/confirm-remove-from-channel.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        username: '',
+      })
+    )
+
+    members.forEach(member => console.log(`TODO: ${member} wants to leave channel, not implemented yet`))
+  }
+
+  const wasWaiting = Container.usePrevious(waiting)
+  React.useEffect(() => {
+    if (wasWaiting && !waiting) {
```
Comment (buoyad) [inline shared/teams/confirm-modals/confirm-remove-from-channel.tsx:52]: this action seems like a nice use for `useRPC`, in which case this nav away could be done more explicitly. But maybe a job for the integration ticket.

File: shared/constants/teams.tsx (hunk 1 above)
Comment (jzila) [reply shared/constants/teams.tsx:45]: Yeah, how about `usernameOrAssertion`?

File: shared/teams/confirm-modals/confirm-remove-from-channel.tsx (hunk 3 above)
Comment (jzila) [reply shared/teams/confirm-modals/confirm-remove-from-channel.tsx:52]: Not sure I follow, we can chat about it in person on the call today.

**Pull request merged**